persistence/src/androidMain/kotlin/com/wire/kalium/persistence/db/UserDatabase.kt

```diff
@@ -39,42 +40,39 @@ sealed interface DatabaseCredentials {
  * that might be necessary for future operations
  * in the future like [nuke]
  */
-internal actual class PlatformDatabaseData(
-    val context: Context,
-    val databaseCredentials: DatabaseCredentials
+actual class PlatformDatabaseData(
+    val context: Context
 )
 
-fun userDatabaseBuilder(
-    context: Context,
+actual fun userDatabaseBuilder(
+    platformDatabaseData: PlatformDatabaseData,
     userId: UserIDEntity,
-    passphrase: UserDBSecret,
-    encrypt: Boolean = true,
-    dispatcher: CoroutineDispatcher
+    passphrase: UserDBSecret?,
+    dispatcher: CoroutineDispatcher,
+    enableWAL: Boolean
 ): UserDatabaseBuilder {
     val dbName = FileNameUtil.userDBName(userId)
 
-    val driver: AndroidSqliteDriver = if (encrypt) {
+    val driver: AndroidSqliteDriver = if (passphrase != null) {
         System.loadLibrary("sqlcipher")
         AndroidSqliteDriver(
             schema = UserDatabase.Schema,
-            context = context,
+            context = platformDatabaseData.context,
             name = dbName,
-            factory = SupportOpenHelperFactory(passphrase.value, true)
+            factory = SupportOpenHelperFactory(
+                passphrase.value,
+                enableWriteAheadLogging = enableWAL
+            )
         )
     } else {
         AndroidSqliteDriver(
             schema = UserDatabase.Schema,
-            context = context,
+            context = platformDatabaseData.context,
             name = dbName,
-            callback = SqliteCallback(UserDatabase.Schema)
+            callback = SqliteCallback(UserDatabase.Schema, enableWAL = enableWAL)
         )
     }
-    val credentials = if (encrypt) {
-        DatabaseCredentials.Passphrase(passphrase)
-    } else {
-        DatabaseCredentials.NotSet
-    }
-    return UserDatabaseBuilder(userId, driver, dispatcher, PlatformDatabaseData(context, credentials))
+    return UserDatabaseBuilder(userId, driver, dispatcher, platformDatabaseData)
 }
 
 fun inMemoryDatabase(
```

Comment:
Should we keep the hardcoded test passphrase or should we randomize it upon creation?
```
val passphrase = "testPass".toByteArray()
```

Reply:
we should keep it